src/kernel/heap.zig

```diff
@@ -303,42 +346,51 @@ fn testBitmapClear(heap: *Heap) !void {
 }
 
 // Copied from std.heap
-fn testAllocator(allocator: *mem.Allocator) !void {
+fn testAllocator(allocator: *std.mem.Allocator) !void {
     var slice = try allocator.alloc(*i32, 100);
     testing.expect(slice.len == 100);
 
-    //for (slice) |*item, i| {
-    //item.* = try allocator.create(i32);
-    //item.*.* = @intCast(i32, i);
-    //}
-    //
-    //for (slice) |item| {
-    //allocator.destroy(item);
-    //}
-    //
+    for (slice) |*item, i| {
+        item.* = try allocator.create(i32);
+        item.*.* = @intCast(i32, i);
+    }
+
+    for (slice) |item| {
+        allocator.destroy(item);
+    }
+
+    slice = allocator.shrink(slice, 50);
+    testing.expect(slice.len == 50);
+    slice = allocator.shrink(slice, 25);
+    testing.expect(slice.len == 25);
+    slice = allocator.shrink(slice, 0);
+    testing.expect(slice.len == 0);
+    slice = try allocator.alloc(*i32, 10);
+    testing.expect(slice.len == 10);
+
     allocator.free(slice);
 
     // The bitmap should now be clear, otherwise free didn't clean up properly
     testBitmapClear(@fieldParentPtr(Heap, "allocator", allocator)) catch unreachable;
 }
 
 // Copied from std.heap
-fn testAllocatorAligned(allocator: *mem.Allocator, comptime alignment: u29) !void {
+fn testAllocatorAligned(allocator: *std.mem.Allocator, comptime alignment: u29) !void {
     // initial
     var slice = try allocator.alignedAlloc(u8, alignment, 10);
     testing.expect(slice.len == 10);
-    // Complex shrinking and re-allocation are not supported yet
+    // Re-allocing isn't supported yet
```

Comment:
Could make these into an issue so not to forget :P

Reply:
Ah yes, I'll do so once this is merged :)